engine/access/ping/engine.go

```diff
@@ -107,21 +107,18 @@ func (e *Engine) startPing() {
 // pingNode pings the given peer and updates the metrics with the result and the additional node information
 func (e *Engine) pingNode(peer *flow.Identity) {
 	id := peer.ID()
-	reachable := e.pingAddress(id)
-	info := e.nodeInfo[id]
-	e.metrics.NodeReachable(peer, info, reachable)
-}
 
-// pingAddress sends a ping request to the given address, and block until either receive
-// a ping respond then return true, or hitting a timeout and return false.
-// if there is other unknown error, return false
-func (e *Engine) pingAddress(target flow.Identifier) bool {
-	// ignore the ping duration for now
-	// ping will timeout in libp2p.PingTimeoutSecs seconds
-	_, err := e.middleware.Ping(target)
+	// ping the node
+	rtt, err := e.middleware.Ping(id) // ping will timeout in libp2p.PingTimeoutSecs seconds
 	if err != nil {
-		e.log.Debug().Err(err).Str("target", target.String()).Msg("failed to ping")
-		return false
+		e.log.Debug().Err(err).Str("target", id.String()).Msg("failed to ping")
+		// report the rtt duration as negative to make it easier to distinguish between pingable and non-pingable nodes
+		rtt = -1
```

Comment:
The only drawback when a node went from normal to unreachable. the diagram would go from 20 (for instance) to -1, which looks like its latency got better, but actually got worse.

is the `-1` would make a node that went from 20 to -1 looked like having reduced latency, but actually is 

For instance the blue line as below
![image](https://user-images.githubusercontent.com/811374/105391150-ac7a0300-5bce-11eb-9b72-bc3d936012e9.png)

An alternative metrics would be:  `1 / rtt`, so that if it's unreachable, we could set it to `0`. Since the ping is usually ranges from `3ms` to `3s`, so the metrics would go from `0.3` to `333`

Thoughts?

Reply:
So few things:
1. On our grafana dashboard we will have one graph which just checks node availability and it will be just checking if the metric value is < 0. We can have another graph which averages the rtt over time
So we may not have a graph which shows both the positive and the negative value.
2. I wanted to avoid floats as much as possible to keep it simple.